src/sql/src/plan/query.rs

```diff
@@ -217,12 +232,30 @@ impl<'a> Fold<Raw, Aug> for NameResolver<'a> {
                 ResolvedObjectName {
                     id: Id::Global(gid),
                     raw_name: normalize::unresolved_object_name(raw_name).unwrap(),
+                    print_id: true,
                 }
             }
         }
     }
 }
 
+pub fn resolve_names_stmt(
+    scx: &StatementContext,
+    query: Statement<Raw>,
+) -> Result<Statement<Aug>, anyhow::Error> {
+    // DNM: I think this is wrong!
+    let qcx = QueryContext::root(scx, QueryLifetime::OneShot);
+    let mut n = NameResolver {
+        status: Ok(()),
+        catalog: qcx.scx.catalog,
+        ctes: HashMap::new(),
+        ids: HashSet::new(),
+    };
+    let result = n.fold_statement(query);
```

Comment:
This looks like it's doing most of the work here? lmk if I need to grok this function to do a more meaningful review!